web-api/tests/filter.rs

```diff
@@ -842,6 +853,121 @@ fn test_filter_date() {
+            )
+            .await?;
+            ingest(
+                &client,
+                &ingestion_url,
+                json!([
+                    { "id": "d1", "snippet": "one" },
+                    { "id": "d2", "snippet": "two", "properties": { "p2": null } },
+                    { "id": "d3", "snippet": "three", "properties": { "p1": [] } }
+                ]),
+            )
+            .await?;
+
+            let bad_request = send_assert_json::<BadRequest>(
+                &client,
+                client
+                    .post(ingestion_url.join("/documents")?)
+                    .json(&json!({ "documents": [
+                        { "id": "d4", "snippet": "four", "properties": { "p1": null } }
+                    ] }))
+                    .build()?,
+                StatusCode::BAD_REQUEST,
+            )
+            .await;
+            assert_eq!(bad_request.kind, Kind::FailedToValidateDocuments);
+            assert_eq!(bad_request.details.ids(), ["d4"].into());
+
+            let documents = send_assert_json::<SemanticSearchResponse>(
+                &client,
+                client
+                    .post(personalization_url.join("/semantic_search")?)
+                    .json(&json!({ "document": { "query": "zero" } }))
+                    .build()?,
+                StatusCode::OK,
+            )
+            .await;
+            assert_eq!(documents.ids(), ["d1", "d2", "d3"].into());
+
+            let documents = send_assert_json::<SemanticSearchResponse>(
+                &client,
+                client
+                    .post(personalization_url.join("/semantic_search")?)
+                    .json(&json!({
+                        "document": { "query": "zero" },
+                        "filter": { "p1": { "$in": ["this", "that"] } }
+                    }))
+                    .build()?,
+                StatusCode::OK,
+            )
+            .await;
+            assert!(documents.is_empty());
+
+            let documents = send_assert_json::<SemanticSearchResponse>(
+                &client,
+                client
+                    .post(personalization_url.join("/semantic_search")?)
+                    .json(&json!({
+                        "document": { "query": "zero" },
+                        "filter": { "p1": { "$in": [] } }
+                    }))
+                    .build()?,
+                StatusCode::OK,
+            )
+            .await;
+            assert!(documents.is_empty());
+
+            let documents = send_assert_json::<SemanticSearchResponse>(
+                &client,
+                client
+                    .post(personalization_url.join("/semantic_search")?)
+                    .json(&json!({
+                        "document": { "query": "zero" },
+                        "filter": { "p2": { "$eq": true } }
+                    }))
+                    .build()?,
+                StatusCode::OK,
```

Comment:
:art: maybe we add an annotation that it should fail once we validate against the schema?